### ISSUE TITLE
`cli-platform` `Stdout.line` alternative without newline suffix?

### ISSUE THREAD

**Issue by JanCVanB** [roc-lang/roc#4120]
The `line` function in `examples/interactive/cli-platform/Stdout` is the only way to write to stdout, and it corresponds to `println!(...)` in Rust. Can we please add a second exposed function in `Stdout` that doesn't append a newline every time? I would use this immediately in https://github.com/JanCVanB/rocbox, but I also expect that other people's CLIs will need return un-newlined output, too.

I don't know what to call it, though... `print` feels lame... `write`? Perhaps the pair should be `write` & `writeLine`?

**Comment by ayazhafiz** [comment]
I agree this would be a good addition. I think “print” is a fine name, it is familiar from many other languages - Python, Rust, Ruby, OCaml, etc

**Comment by JanCVanB** [comment]
Yes, it is familiar. Should we make the pair `Stdout.print` & `Stdout.printLine` instead of `Stdout.print` & `Stdout.line`?

**Comment by rtfeldman** [comment]
The more often a function is called, the bigger the relative downside a long name is (and the more easily people will learn a shorter name even if it's less self-descriptive). I named it `.line` with this in mind; it's definitely going to be the most frequently called function in the module, so giving it a short name seems very worthwhile!

**Comment by rtfeldman** [comment]
As for `print` vs `write`, I would go with `write`.

Maybe because of my early QBASIC experience, I've always been sensitive to the fact that in some languages, `print` adds a newline at the end, and others don't.

In contrast, I don't know of any `write` ever adds a newline at the end!

**Comment by JanCVanB** [comment]
I like that better. Here are the two contenders written out:

Proposal A:
```
Stdout.printLine "foo"
Stdout.print "bar"
```

Proposal B:
```
Stdout.line "foo"
Stdout.write "bar"
```

**Comment by JanCVanB** [comment]
When writing those out, I noticed something else - if we renamed the newline-ful function to be longer than the newline-less function, it will encourage many people to use the shorter function name out of sheer minimalism/reflex/idk. I initially wrote Proposal A with `print` first, just because my brain sorted them shorter to longer... only after writing Proposal B did I go back and reorder Proposal A to fit the sorting of common to uncommon. Newline-less will be WAY less common than newline-ful, so now I'm a big fan of Proposal B.
